torchgeo/datasets/geo.py

```diff
@@ -388,13 +399,22 @@ def __init__(
                     mint: float = 0
                     maxt: float = sys.maxsize
                     if "date" in match.groupdict():
-                        date = match.group("date")
+                        date: str = match.group("date")
+                        dates.append((os.path.basename(filepath), date))
                         mint, maxt = disambiguate_timestamp(date, self.date_format)
 
                     coords = (minx, maxx, miny, maxy, mint, maxt)
                     self.index.insert(i, coords, filepath)
                     i += 1
 
+        if self.as_time_series:
```

Comment:
I kept this for development/debugging purposes but also might be nice to have access to the available dates of files when working with time series data. Maybe as a dict so there is no dependency on pandas?